lib/setup.c

```diff
@@ -921,6 +926,40 @@ ___SIZE_TS requested_words_still;)
 }
 
 
+___EXP_FUNC(___BOOL, ___fdset_resize)
+   ___P((int fd1,
+         int fd2),
+        (fd1,
+         fd2)
+int fd1;
+int fd2;)
+{
+#ifdef USE_select_or_poll
+
+  ___processor_state ___ps = ___PSTATE;
+  int fd = (fd2 > fd1) ? fd2 : fd1;
+
+  if (fd < ___ps->os.fdset.size)
+    return 0;
+
+  ___fdset_resize_heap_overflow_clear ();
+
+  ___sync_op_struct sop;
+
+  sop.op = OP_FDSET_RESIZE;
+  sop.arg[0] = fd;
+
+  on_all_processors (___PSP &sop);
+
+  return ___fdset_resize_heap_overflow ();
```

Comment:
Cleared the negations in pseudo for simplicity.
Basically what we want is to count how many resizes are being serialized and avoid clobbering overflow on the second start.

If after serialization they get combined into a single execution (`MAX_COMBINING`), then `__fdset_resize_end()` is completely unnecessary and we can simply clear the resize count.